### PR TITLE
Add unit tests for namespaces

### LINKED ISSUE #11562: Create tests if this is necessary additional tests to  `FIMDBHelper`
|EPIC|Component|
|---|---|
| https://github.com/wazuh/wazuh/issues/9103 | FIM |

### Description

This issue intends to add tests to assure that the new methods is working properly. It implies:
- [x] Create unit tests for FimDBUtils namespace
- [x]  Add if this were unit tests needed for FimDBHelper namespace

### PR DESCRIPTION
**Pull request by FrancoRivero**
|EPIC|Component|
|---|---|
| https://github.com/wazuh/wazuh/issues/9103 | FIM |

### Description

This issue intends to add tests to assure that the new methods is working properly. It implies:
- [x] Create unit tests for FimDBUtils namespace
- [x]  Add if this were unit tests needed for FimDBHelper namespace

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
- [x] Unit tests passed
- [x] Review logs syntax and correct language

- Memory tests for Linux
  - [x] Scan-build report
  - [x] Valgrind (memcheck and descriptor leaks check)
  - [x] cppcheck (static code analysis)
  - [x] Coverage (unit tests coverage code)
  - [x] Sformat (code style)

Closes #11562

### CODE REVIEW

File: src/syscheckd/db/tests/db/FIMDB/fimDBUtils/fimDBUtilsTest.cpp [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    try
+    {
+        const auto paths = FimDBUtils::getPathsFromPattern<FIMDBMOCK>("test.txt");
+    }
+    catch (const std::exception& ex)
+    {
+        std::cout << ex.what() << std::endl;
+        ASSERT_EQ(ex.what(), "ERROR");
+    }
+}
```
Comment (antoniomanuelfr) [inline src/syscheckd/db/tests/db/FIMDB/fimDBUtils/fimDBUtilsTest.cpp:140]: Add a newline to the end of the file
```suggestion
}

```

File: src/syscheckd/db/src/fimDBUtils.hpp [hunk 2]
```diff
@@ -21,16 +44,18 @@ namespace FimDBUtils
...
     {
         std::vector<std::string> paths;
         nlohmann::json resultQuery;
 
         try
         {
-            const auto filter { std::string("WHERE inode=") + std::to_string(inode) + std::string(" AND dev=") + std::to_string(dev) };
-            const auto query = FIMDBHelper::dbQuery(FIMBD_FILE_TABLE_NAME, FILE_PRIMARY_KEY, filter, FILE_PRIMARY_KEY);
-            FIMDBHelper::getDBItem<FIMDB>(resultQuery, query);
+            const auto filter = std::string("WHERE inode=") + std::to_string(inode) + std::string(" AND dev=") + std::to_string(dev);
```
Comment (JoseAntonioMHerrera) [inline src/syscheckd/db/src/fimDBUtils.hpp:55]: I think you can get rid of the construction of strings objects for "WHERE inode=" and " AND dev=" as you are already creating strings objects with the std::to_string() for concatenation purposes.

File: src/syscheckd/db/tests/db/FIMDB/fimDBWrapper/fimDBWrapperTest.cpp [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    int count = 0;
+    EXPECT_CALL(FIMDBMOCK::getInstance(), executeQuery(testing::_, testing::_))
+    .Times(testing::AtLeast(1))
+    .WillOnce(testing::InvokeArgument<1>(ReturnTypeCallback::SELECTED, R"({"count":5})"_json));
+    EXPECT_NO_THROW(
+    {
+        count = FIMDBHelper::getCount<FIMDBMOCK>(tableName);
+    });
+    ASSERT_NE(count, 0);
+    ASSERT_EQ(count, 5);
```
Comment (JoseAntonioMHerrera) [inline src/syscheckd/db/tests/db/FIMDB/fimDBWrapper/fimDBWrapperTest.cpp:94]: I think you can delete the assertion count not equal to 0, the EXPECT_NO_THROW and ASSERT_EQ(count, 5) statements are already covering all the possible cases defined for that test.

File: src/syscheckd/db/tests/db/FIMDB/fimDBWrapper/fimDBWrapperTest.cpp [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    const auto countQuery = FimDBUtils::dbQuery(tableName, query, "", "");
+    EXPECT_CALL(FIMDBMOCK::getInstance(), executeQuery(testing::_, testing::_))
+    .Times(testing::AtLeast(1))
+    .WillOnce(testing::InvokeArgument<1>(ReturnTypeCallback::SELECTED, R"({"count":2})"_json));
+    EXPECT_NO_THROW(
+    {
+        count = FIMDBHelper::getCount<FIMDBMOCK>(tableName, query);
+    });
+    ASSERT_NE(count, 0);
+    ASSERT_EQ(count, 2);
```
Comment (JoseAntonioMHerrera) [inline src/syscheckd/db/tests/db/FIMDB/fimDBWrapper/fimDBWrapperTest.cpp:113]: Sames as before, you can delete ASSERT_NE(count, 0)

**Review by JoseAntonioMHerrera — COMMENTED** [review]
Looks good, some minor things to discuss

File: src/syscheckd/db/src/fimDBUtils.hpp (hunk 2 above)
Comment (FrancoRivero) [reply src/syscheckd/db/src/fimDBUtils.hpp:55]: You are right, I'm changing that ! Thank you !

File: src/syscheckd/db/tests/db/FIMDB/fimDBWrapper/fimDBWrapperTest.cpp (hunk 4 above)
Comment (FrancoRivero) [reply src/syscheckd/db/tests/db/FIMDB/fimDBWrapper/fimDBWrapperTest.cpp:111]: You are right, this is a redundant validation. I'm changing that ! Thank you !

File: src/syscheckd/db/tests/db/FIMDB/fimDBWrapper/fimDBWrapperTest.cpp (hunk 3 above)
Comment (FrancoRivero) [reply src/syscheckd/db/tests/db/FIMDB/fimDBWrapper/fimDBWrapperTest.cpp]: You are right, this is a redundant validation. I'm changing that ! Thank you !

**Review by JoseAntonioMHerrera — APPROVED** [review]
LGTM!

File: src/syscheckd/db/include/db.h [hunk 5]
```diff
@@ -63,28 +63,34 @@ void fim_db_init(int storage,
  * @param file_path File path can be a pattern or a primary key
  * @param data Pointer to the data structure where the callback context will be stored.
  *
- * @return FIMDB_OK on success, FIMDB_ERROR on failure.
+ * @retval FIMDB_OK on success.
+ * @retval FIMDB_FULL if the table limit was reached.
+ * @retval FIMDB_ERR on failure.
  */
-int fim_db_get_path(const char* file_path, callback_context_t data);
+FIMDBErrorCodes fim_db_get_path(const char* file_path, callback_context_t data);
```
Comment (Dwordcito) [inline src/syscheckd/db/include/db.h:70]: It would be better if it is singular, eg FIMDBErrorCode

File: src/syscheckd/db/include/db.h [hunk 6]
```diff
@@ -63,28 +63,34 @@ void fim_db_init(int storage,
...
  * @param pattern Pattern to be searched.
  * @param data Pointer to the data structure where the callback context will be stored.
  *
- * @return FIMDB_OK on success, FIMDB_ERROR on failure.
+ * @retval FIMDB_OK on success.
+ * @retval FIMDB_FULL if the table limit was reached.
+ * @retval FIMDB_ERR on failure.
  */
-int fim_db_file_pattern_search(const char* pattern, callback_context_t data);
+FIMDBErrorCodes fim_db_file_pattern_search(const char* pattern, callback_context_t data);
```
Comment (Dwordcito) [inline src/syscheckd/db/include/db.h:82]: ditto

File: src/syscheckd/db/include/db.h [hunk 7]
```diff
@@ -63,28 +63,34 @@ void fim_db_init(int storage,
...
  *
  * @param path Path of the entry to be removed.
  *
- * @return FIMDB_OK on success, FIMDB_ERR otherwise.
+ * @retval FIMDB_OK on success.
+ * @retval FIMDB_FULL if the table limit was reached.
+ * @retval FIMDB_ERR on failure.
  */
-int fim_db_remove_path(const char* path);
+FIMDBErrorCodes fim_db_remove_path(const char* path);
```
Comment (Dwordcito) [inline src/syscheckd/db/include/db.h:93]: ditto

File: src/syscheckd/db/include/db.h [hunk 8]
```diff
@@ -107,9 +113,12 @@ int fim_db_get_count_file_entry();
...
  * @param updated The updated is a flag to keep if the operation was updated or not.
  * @return The result of the update operation.
- * @retval Returns any of the values returned by fim_db_set_scanned and fim_db_insert_entry.
+ *
+ * @retval FIMDB_OK on success.
+ * @retval FIMDB_FULL if the table limit was reached.
+ * @retval FIMDB_ERR on failure.
  */
-int fim_db_file_update(const fim_entry* data, bool* updated);
+FIMDBErrorCodes fim_db_file_update(const fim_entry* data, bool* updated);
```
Comment (Dwordcito) [inline src/syscheckd/db/include/db.h:121]: ditto

File: src/syscheckd/db/src/file.cpp [hunk 9]
```diff
@@ -16,10 +16,10 @@
 extern "C" {
 #endif
 
-int fim_db_file_pattern_search(const char* pattern, callback_context_t callback)
+FIMDBErrorCodes fim_db_file_pattern_search(const char* pattern, callback_context_t callback)
```
Comment (Dwordcito) [inline src/syscheckd/db/src/file.cpp:19]: ditto

File: src/syscheckd/db/src/file.cpp [hunk 10]
```diff
@@ -39,7 +39,7 @@ int fim_db_file_pattern_search(const char* pattern, callback_context_t callback)
     return retVal;
 }
 
-int fim_db_get_path(const char* file_path, callback_context_t callback)
+FIMDBErrorCodes fim_db_get_path(const char* file_path, callback_context_t callback)
```
Comment (Dwordcito) [inline src/syscheckd/db/src/file.cpp:42]: ditto

File: src/syscheckd/db/src/file.cpp [hunk 11]
```diff
@@ -99,7 +98,7 @@ int fim_db_get_path(const char* file_path, callback_context_t callback)
 }
 
 
-int fim_db_remove_path(const char* path)
+FIMDBErrorCodes fim_db_remove_path(const char* path)
```
Comment (Dwordcito) [inline src/syscheckd/db/src/file.cpp:101]: ditto

File: src/syscheckd/db/src/file.cpp [hunk 12]
```diff
@@ -159,7 +158,7 @@ int fim_db_get_count_file_entry()
     return count;
 }
 
-int fim_db_file_update(const fim_entry* data, bool* updated)
+FIMDBErrorCodes fim_db_file_update(const fim_entry* data, bool* updated)
```
Comment (Dwordcito) [inline src/syscheckd/db/src/file.cpp:161]: ditto

**Review by Dwordcito — COMMENTED** [review]
minimal change

**Pull request merged**